### PR TITLE
Add Apache 2 license header to every Go file

### PR DESCRIPTION
**Pull request by ytsarev**
* Add Apache 2 headers
* Add incredible piece of automation for header injection

### CODE REVIEW

File: hack/header_file [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
```
Comment (k0da) [inline hack/header_file:1]: Can't we sync it with ./hack/boilerplate.txt and use same header here and in controller-gen?

**Comment by kuritka** [comment]
I dind't study deepply, but there are tools doing that:  https://github.com/cloudposse/copyright-header. Maybe we can manipulate with headers before GOLINT

File: hack/header_file (hunk 1 above)
Comment (ytsarev) [reply hack/header_file]: @k0da thanks, i've made boilerplate a single source of headers file

**Comment by ytsarev** [comment]
@kuritka I definitely don't want to employ super sophisticated machinery with lots of rubygem dependencies here.

Also tried https://github.com/google/addlicense . It does pretty much the same stuff as the tiny shell script provided.

So I suggest we stick with the simplest version of it which does not bring in any new dependencies

**Review by kuritka — DISMISSED** [review]

**Review by kuritka — APPROVED** [review]

**Review by somaritane — APPROVED** [review]

**Pull request merged**